src/System.Linq.Expressions/src/System/Linq/Expressions/NewExpression.cs

```diff
@@ -250,15 +250,15 @@ private static void ValidateNewArgs(ConstructorInfo constructor, ref ReadOnlyCol
                 for (int i = 0, n = arguments.Count; i < n; i++)
                 {
                     Expression arg = arguments[i];
-                    RequiresCanRead(arg, "argument");
+                    RequiresCanRead(arg, $"{nameof(arguments)}[{i}]");
```

Comment:
Same as before; quite a lot of string allocations due to interpolation. I'd stick with just the parameter name and omit the indexing (which is also rather uncommon in exception properties).